File: es/01-programming/csharp/ejercicios/nivel-05-experto/ejercicio-02-servidor-http.cs
using System;

public class ServidorHttp
{
    private readonly int _puerto;
    public int Puerto => _puerto;

    public ServidorHttp(int puerto)
    {
        _puerto = puerto;
    }

    public string ProcesarRuta(string ruta)
    {
        // TODO: enruta la petición sin red:
        //   "/"                -> "<h1>Inicio</h1>"
        //   "/saludo"          -> "<h1>Hola, bienvenido</h1>"
        //   "/saludo?nombre=X" -> "<h1>Hola, X</h1>"
        //   "/api/hora"        -> DateTime.Now.ToString("HH:mm:ss")
        //   cualquier otra     -> "404 No encontrado"
        throw new NotImplementedException("TODO: implementar ProcesarRuta(string)");
    }
}

public static class Ejercicio02
{
    public static ServidorHttp CrearServidor(int puerto)
    {
        // TODO: devuelve new ServidorHttp(puerto).
        throw new NotImplementedException("TODO: implementar CrearServidor(int)");
    }
}

File: es/01-programming/javascript/ejercicios/nivel-05-experto/ejercicio-02-servidor-http.js
const http = require("node:http");

const PUERTO = 3000;

const productos = [
  { nombre: "Laptop", precio: 800 },
  { nombre: "Mouse", precio: 20 },
];

function responder(res, codigo, tipo, cuerpo) {
  // TODO: envía headers (Content-Type y Content-Length) y el cuerpo.
  throw new Error("TODO: implementar responder(...)");
}

function crearServidor() {
  // TODO: devuelve un http.Server que responde por ruta:
  // "/" 200 texto; "/hora" 200 ISO; "/saludo?nombre=" 200; "/productos" 200 JSON; otra 404.
  throw new Error("TODO: implementar crearServidor()");
}

if (require.main === module) {
  crearServidor().listen(PUERTO, () => {
    console.log(`Servidor escuchando en http://localhost:${PUERTO}`);
  });
}

module.exports = { crearServidor, responder, PUERTO, productos };
